daemon.c

```diff
@@ -687,6 +691,32 @@ static void mountprog_3(struct svc_req *rqstp, register SVCXPRT * transp)
     return;
 }
 
+static int
+_socket_getdomain(int socket, int *domain, socklen_t *len)
+{
+#ifdef SO_DOMAIN
+    return getsockopt(socket, SOL_SOCKET, SO_DOMAIN, &domain, &len);
+#else
+# ifdef PROC_PIDFDSOCKETINFO
```

Comment:
You can get rid of the nesting if you use `defined()` instead:

```suggestion
#elif defined(PROC_PIDFDSOCKETINFO)
```

Reply:
I contributed TIRPC patches upstream for macOS and they were integrated; there is one more edit needed which I need to revisit.